### PR TITLE
WiP/Discussion: Method grouping and method signatures

### PR DESCRIPTION
**Pull request by leggetter**
Let's use this PR to discuss and try out ideas. Once agreed we'll create a new PR (or rebase) with the changes reflecting what was agreed.

The two commits show different changes:
- https://github.com/Nexmo/client-library-specification/commit/2b1c0f67ebd81662dec4c2ac4945d03787ab2a72 a suggested regrouping around what seems to be associated functionality
- https://github.com/Nexmo/client-library-specification/commit/bbf89f163dd49dd2d24597b9e52764cc31a2de57 namespaces and method names. In some cases parameters have been provided. Specifically to group what feels like search functionality or varying levels of the same functionality.

I've also suggested that some of the messaging functionality should sit under the `sms` namespace. `messaging` seems very generic and may cause confusion in future. The Node library provides specific methods addressing specific message types. Maybe that's a consideration too?

### CODE REVIEW

**Comment by tjlytle** [comment]
Only thing that sticks out to me at this point is the 'pricing' under numbers, that's (currently) pricing for outbound messages, but should probably include voice rates as well in the future. Either way, I think that makes sense as it's own top level namespace.

**Comment by LisaRoskam** [comment]
A small note in regards to the Product namespaces. Please let's stick to 
the standard terms used by the Product team to prevent confusion. They 
have chosen to use the over-arching umbrella term Messaging to include:
SMS API
US Short Codes API
Chat App API
Conversion API

See the Nexmo Product Catalogue page in the internal wiki - 
https://nexmoinc.atlassian.net/wiki/display/PROD/Nexmo+Product+Catalogue

> Phil Leggetter <mailto:notifications@github.com>
> May 20, 2016 at 7:15 PM
>
> Let's use this PR to discuss and try out ideas. Once agreed we'll 
> create a new PR (or rebase) with the changes reflecting what was agreed.
>
> The two commits show different changes:
>
>   * 2b1c0f6
>     <https://github.com/Nexmo/client-library-specification/commit/2b1c0f67ebd81662dec4c2ac4945d03787ab2a72>
>     a suggested regrouping around what seems to be associated
>     functionality
>   * bbf89f1
>     <https://github.com/Nexmo/client-library-specification/commit/bbf89f163dd49dd2d24597b9e52764cc31a2de57>
>     namespaces and method names. In some cases parameters have been
>     provided. Specifically to group what feels like search
>     functionality or varying levels of the same functionality.
>
> I've also suggested that some of the messaging functionality should 
> sit under the |sms| namespace. |messaging| seems very generic and may 
> cause confusion in future. The Node library provides specific methods 
> addressing specific message types. Maybe that's a consideration too?
>
> ------------------------------------------------------------------------
>
>
>         You can view, comment on, or merge this pull request online at:
>
> https://github.com/Nexmo/client-library-specification/pull/8
>
>
>         Commit Summary
>
>   * suggested updates to method grouping
>   * suggested method nameing and in some cases signature
>
>
>         File Changes
>
>   * *M* templates/README.md
>     <https://github.com/Nexmo/client-library-specification/pull/8/files#diff-0>
>     (63)
>
>
>         Patch Links:
>
>   * https://github.com/Nexmo/client-library-specification/pull/8.patch
>   * https://github.com/Nexmo/client-library-specification/pull/8.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/Nexmo/client-library-specification/pull/8>
>

-- 
EMEA Support Manager
Nexmo, Inc. | 44 Featherstone St, 1stFloor | London, EC1V 8RN | United 
Kingdom
+44 115 70 70 707| help.nexmo.com <http://help.nexmo.com/>

File: templates/README.md [hunk 1]
```diff
@@ -74,43 +74,47 @@ API Coverage
...
 * Number Insight
-    * [ ] Basic
-    * [ ] Standard
-    * [ ] Advanced
-    * [ ] Webhook Notification
+    * [ ] Basic (`numberInsight({level:basic})`)
+    * [ ] Standard (`numberInsight({level:standard})`)
+    * [ ] Advanced (`numberInsight({level:advanced})`)
+    * [ ] Check Webhook Notification Signature (`webhook.validate`)
+* Pricing
```
Comment (leggetter) [inline templates/README.md:93]: @tjlytle - pricing now has own namespace

**Comment by tjlytle** [comment]
@leggetter Do the changes in #15 resolve this?

**Comment by lornajane** [comment]
closing due to inactivity

**Pull request closed without merging**